scipy/optimize/_shgo.py

```diff
@@ -77,12 +81,11 @@ def shgo(func, bounds, args=(), constraints=None, n=None, iters=1,
 
     n : int, optional
         Number of sampling points used in the construction of the simplicial
-        complex. Note that this argument is only used for ``sobol`` and other
-        arbitrary `sampling_methods`. In case of ``sobol``, it must be a
-        power of 2: ``n=2**m``, and the argument will automatically be
-        converted to the next higher power of 2. Default is 100 for
-        ``sampling_method='simplicial'`` and 128 for
-        ``sampling_method='sobol'``.
+        complex. For the default ``simplicial`` sampling method 2**dim + 1
+        sampling points are generated instead of the default `n=100`. For all
+        other specified values `n` sampling points are generated. For
+        ``sobol``, ``halton`` and other arbitrary `sampling_methods` `n=100` or
+         another speciefied number of sampling points are generated.
```

Comment:
```suggestion
        another speciefied number of sampling points are generated.
```

This should fix the docstring error.

Reply:
Thank you.